docs/source/55_api.md

```diff
@@ -339,6 +341,138 @@ val bcastFeatures = sc.loadFeatures("my/features.adam").broadcast()
+automatically dropped, providing the filter we are interested in. This query is
+useful for trying to identify genotypes that overlap features of interest. For
+example, if our feature file contains all the exonic regions of the genome, we
+could perform this query to extract all genotypes that fall in exonic regions.
+
+```scala
+// Inner join will filter out genotypic data not represented in the feature dataset
+val genotypes = sc.loadGenotypes(“my/genotypes.adam”)
+val features = sc.loadFeatures(“my/features.adam”)
+
+// We can use ShuffleRegionJoin…
+val joinedGenotypesShuffle = genotypes.shuffleRegionJoin(features)
+
+// …or BroadcastRegionJoin
+val joinedGenotypesBcast = features.broadcastRegionJoin(genotypes)
+
+// In the case that we only want Genotypes, we can use a simple predicate
+val filteredGenotypesShuffle = joinedGenotypesShuffle.rdd.map(_._1)
+
+val filteredGenotypesBcast = joinedGenotypesBcast.rdd.map(_._2)
+```
+
+After the join, we can perform a predicate function on the resulting RDD to
+manipulate it into providing the answer to our question. Because we were
+interested in only getting the Genotypes that overlap the features, we used a
+predicate.
+
+Notice that at the end of the join, we can access the RDD resulting from the
+join and perform a map operation on it. This `map` call can, in fact, be
+replaced with any predicate function.
+
+###### Group overlapping variant data by the gene they overlap
+
+This query joins an RDD of Variants against an RDD of Features, and immediately
+performs a group-by on the Feature. This produces an RDD whose elements are a
+tuple containing a Feature, and all of the Variants overlapping the Feature.
+This query is useful for trying to identify annotated variants that may
+interact (identifying frameshift mutations within a transcript that may act as
+a pair to shift and then restore the reading frame) or as the start of a query
+that computes variant density over a set of genomic features.
+
+```scala
+// Inner join with a group by on the features
+val features = sc.loadFeatures(“my/features.adam”)
+val variants = sc.loadVariants(“my/variants.adam")
+
+// As a ShuffleRegionJoin, it can be implemented as follows:
+val variantsByFeatureShuffle = features.shuffleRegionJoinAndGroupByLeft(variants)
+
+// As a BroadcastRegionJoin, it can be implemented as follows:
+val variantsByFeatureBcast = variants.broadcastRegionJoinAndGroupByRight(features)
+```
+
+Notice that the type of join determines which dataset calls the join and which
+gets passed as an argument. In the ShuffleRegionJoin, `features` calls the
+join, but in BroadcastRegionJoin, `variants` calls the join.
+
+This distinction is very important to understanding the difference between
+BroadcastRegionJoin and ShuffleRegionJoin. It is important to first understand
+the difference between what is called the "left" and "right" datasets. Simply
+put, the left dataset is the dataset that calls the join, and the right dataset
```

Comment:
I don't like this as is written because it this isn't wholly consistent throughout ADAM due to implementation necessities. E.g., the `broadcastJoinAgainst` operator uses the right side of the join as the caller and the left side of the join as the parameter.

I would also omit explaining right vs. left in a join and possibly refer out to some database reference material (e.g., a good textbook).